src/main/java/com/ibm/cics/bundle/parts/BundlePublisher.java

```diff
@@ -111,28 +100,35 @@ public void addResource(BundleResource resource) throws PublishException {
 	 * @return a Define statement to add to cics.xml if the file was a bundle part, or null if not
 	 */
 	private static Define getDefine(Path toImport) {
-		String fileName = toImport.getFileName().toString();
-		
-		int dot = fileName.lastIndexOf(".");
-		if (dot <= 0) {
-			log.debug("Couldn't determine bundle part type for file name: " + fileName);
-			return null;
-		}
-		
-		String extension = fileName.substring(dot + 1).toLowerCase();
-		BundlePartType type = BundlePartType.getType(extension);
-		if(type != null) {
-			String name = fileName.substring(0, dot);
-			if (name.length() > 0) {
-				return new Define(name, type, toImport.toString());
-			} else {
-				log.debug("Couldn't determine bundle part name for file \"" + fileName + "\"");
-				return null;
+		if (toImport != null) {
```

Comment:
Presumably the right thing to do in this situation is Objects.requireNonNull(toImport)?  A null parameter to this method isn't meaningful, and throwing an NPE is probably the right thing to do (which would've happened anyway I suppose).

Reply:
Fixed